deploy.py

```diff
@@ -447,7 +447,11 @@ def provision_execute(name):
     for stage in stages:
         logging.info(f"Running stage")
         results = []
-        with concurrent.futures.ThreadPoolExecutor() as executor:
+        max_executors = len(hosts)
+        # The last stage should be synchronous
```

Comment:
We should add the `why` to this sentence to understand in the future the logic about this